nixos/modules/services/hardware/amdgpu.nix

```diff
@@ -10,11 +10,27 @@ in {
       series cards. Note: this removes support for analog video outputs,
       which is only available in the `radeon` driver
     '';
-    initrd.enable = lib.mkEnableOption ''
-      loading `amdgpu` kernelModule in stage 1.
-      Can fix lower resolution in boot screen during initramfs phase
-    '';
-    opencl.enable = lib.mkEnableOption ''OpenCL support using ROCM runtime library'';
+    enable = lib.mkEnableOption "AMD GPU";
```

Comment:
I don't see the point in this enable option. It's not at all obvious that it enables initrd and opencl. I think it'd be better to simply let users pick and choose what they like.

The existence of an enable option also implies that it's necessary to enable to have AMDGPU support which is not at all true; it merely enables some optional AMDGPU features.